File: src/main/java/br/com/leonardoferreira/jirareport/domain/vo/changelog/JiraChangelogItem.java
package br.com.leonardoferreira.jirareport.domain.vo.changelog;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.time.LocalDateTime;
import lombok.Data;

/**
 * @author s2it_leferreira
 * @since 11/06/18 13:48
 */
@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class JiraChangelogItem {

    private String field;

    private String fieldtype;

    private String from;

    private String fromString;

    private String to;

    private String toString;

    @JsonIgnore
    private LocalDateTime created;

}

File: src/main/kotlin/br/com/jiratorio/domain/jira/changelog/JiraChangelogItem.kt
package br.com.jiratorio.domain.jira.changelog

import br.com.jiratorio.extension.toStringBuilder
import com.fasterxml.jackson.annotation.JsonIgnore
import java.time.LocalDateTime

data class JiraChangelogItem(

    val field: String? = null,

    val fieldtype: String? = null,

    val from: String? = null,

    val fromString: String? = null,

    val to: String? = null,

    val toString: String? = null,

    @JsonIgnore
    var created: LocalDateTime? = null

) {

    override fun toString(): String =
        toStringBuilder(
            JiraChangelogItem::field,
            JiraChangelogItem::fieldtype
        )

}
